File: EntitiesVb/CurrentAccount.vb
Public Class CurrentAccount
    Inherits Account
    Private _OverdraftAmount As String
    Public Property OverdraftAmount() As String
        Get
            Return _OverdraftAmount
        End Get
        Set(ByVal value As String)
            _OverdraftAmount = value
        End Set
    End Property
    Public Overrides Sub Extraer(value As Decimal)
        If isSaldoSuficiente(value) Then
            _saldo -= value
        End If
    End Sub
    Private Function isSaldoSuficiente(monto As Decimal) As Boolean
        Return Saldo >= monto - OverdraftAmount
    End Function
    Public Overrides Function ToString() As String
        Return "Current Account: " & Number
    End Function
End Class

File: EntitiesCs/CurrentAccount.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace EntitiesCs
{
    public class CurrentAccount : Account
    {
        public CurrentAccount(int number, decimal saldo, decimal overdraftAmount) : base(number, saldo)
        {
            OverdraftAmount = overdraftAmount;
        }
        public decimal OverdraftAmount { get; set; }
        public override void Extraer(decimal value)
        {
            if (isSaldoSuficiente(value)) // validación
                saldo -= value;
        }
        private bool isSaldoSuficiente(decimal Monto)
        {
            return Saldo >= Monto - OverdraftAmount;
        }
    }
}
